sky/engine/core/dart/window.dart

```diff
@@ -18,41 +24,102 @@ class WindowPadding {
   final double left;
 }
 
+/// An identifier used to select a user's language and formatting preferences,
+/// consisting of a language and a country. This is a subset of locale
+/// identifiers as defined by BCP 47.
 class Locale {
-  Locale(this.languageCode, this.countryCode);
+  const Locale(this.languageCode, this.countryCode);
 
+  /// The primary language subtag for the locale.
   final String languageCode;
+
+  /// The region subtag for the locale.
   final String countryCode;
 
+  bool operator ==(dynamic other) {
```

Comment:
Language support could help us.  The VM could provide an extensional equals function that's a dual to `identical`.